api-specificatie/Bevraging-Ingeschreven-Persoon/openapi.yaml

```diff
@@ -2042,3 +2182,280 @@ components:
+    Bewoning_embedded:
+      type: "object"
+      properties:
+        bewoners:
+          title: "bewoondDoor"
+          type: "array"
+          description: ""
+          minItems: 1
+          items:
+            $ref: "#/components/schemas/BewonerHal"
+    BewonerHal:
+      allOf:
+        - $ref: '#/components/schemas/Bewoner'
+        - type: "object"
+          properties:
+            _links:
+              $ref: "#/components/schemas/Bewoner_links"
+    Bewoner:
+      type: "object"
+      description: "<body><p> <strong>Dit betreft een historische resource. De BEWONER\
+        \ kan in het verleden een ingeschreven persoon zijn geweest, maar nu niet\
+        \ meer ingeschreven zijn</strong> </p><p>\n <em>*</em> datumIngangGeldigheid**\
+        \ : Datum waarop de gegevens over de bewoning geldig zijn geworden.</p><p>\n\
+        \ <em>*</em> datumEindeGeldigheid** : Datum waarop de geldigheid van de gegevens\
+        \ over de bewoning beeindigd is.</p></body>"
+      properties:
+        burgerservicenummer:
+          type: "string"
+          title: "Burgerservicenummer"
+          description: "<body>Het burgerservicenummer, bedoeld in artikel 1.1 van\
+            \ de Wet algemene bepalingen burgerservicenummer.</body><body>Elk ingeschreven\
+            \ natuurlijk persoon heeft een BSN, een nummer dat de natuurlijk persoon\
+            \ uniek identificeert in overheidsadministraties. De attribuutsoort kent\
+            \ onder meer historie omdat het mogelijk is dat het Burgerservicenummer\
+            \ van een natuurlijk persoon wijzigt, met name vanwege ambtelijke correcties.\
+            \ Zie verder de toelichting in de GBA</body><body>Alle nummers waarvoor\
+            \ geldt dat, indien aangeduid als (s0 s1 s2 s3 s4 s5 s6 s7 s8), het resultaat\
+            \ van (9*s0) + (8*s1) + (7*s2) +...+ (2*s7) - (1*s8) deelbaar is door\
+            \ elf. Er moeten dus 9 cijfers aanwezig zijn.</body>"
+          maxLength: 9
+          minLength: 9
+          example: "555555914"
+        geheimhoudingPersoonsgegevens:
+          type: "boolean"
+          title: "Indicatie geheim"
+          description: "<body>Een aanduiding die aangeeft dat gegevens wel of niet\
+            \ verstrekt mogen worden.</body><body><p>Een aanduiding die aangeeft dat\
+            \ gegevens wel of niet verstrekt mogen worden. Indien true: op verzoek\
+            \ van deze persoon is het verstrekken van gegevens over deze persoon aan\
+            \ bepaalde derden niet toegestaan.</p></body><body></body><body>â€œj(a)â\
+            €? of leeg</body>"
+        datumEindeGeldigheid:
+          $ref: "#/components/schemas/Datum_onvolledig"
+        datumIngangGeldigheid:
+          $ref: "#/components/schemas/Datum_onvolledig"
+        geboortedatum:
+          $ref: "#/components/schemas/Geboortedatum"
+        naam:
+          $ref: "#/components/schemas/Naam"
+    Geboortedatum:
```

Comment:
Ik blijf dit een gekunstelde en implementatie gerichte keuze vinden (we zijn een standaard aan het maken. Hoe los je dat op als er straks een endpoint is dat alleen een geboorteland nodig heeft ? Land en plaats zijn optioneel binnen geboorte en m.b.v. de fields parameter zou je zelfs kunnen kiezen om die niet op te vragen als je alleen de geboortedatum wil. 
Ik snap ook dat we voor consumergemak gaan, maar wil wel waken voor wildgroei op dit soort constructies die binnen verschillende API's uit elkaar kunnen gaan lopen. Of zien we dat niet als een probleem ?

Reply:
Het is niet zo zwart/wit zoals je het nu schetst, als ik het begrijp. De keuze die afgelopen vrijdag volgens mij is gemaakt om te kijken of het mogelijk is om meer hergebruik toe te passen.
  
Een manier die toen is besproken/geschetst, is om te extenden (een (sub)resource laten afleiden van een andere), maar het gebruiken van een sub(resource) als property is ook een manier.

Ik denk dat we per situatie moet kijken wat de handigste manier is om te hergebruiken **en** of het kan, in plaats van 1 oplossing te bedenken en daaraan vasthouden. Dat is denk ik wat je bedoel met de opmerking om geboorte te hergebruiken en met fields te filteren als je alleen geboorteland wil. Want hiermee dwing je eigenlijk een provider om, in dit geval geboortedatum te kunnen leveren, terwijl hij dat mogelijk niet heeft